ouroboros-network/src/Ouroboros/Network/BlockFetch/Client.hs

```diff
@@ -0,0 +1,345 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE RecordWildCards            #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+
+module Ouroboros.Network.BlockFetch.Client (
+    blockFetchClient,
+    FetchClientPolicy(..),
+
+    FetchClientRegistry(..),
+    newFetchClientRegistry,
+    bracketFetchClient,
+  ) where
+
+import           Data.Maybe
+import qualified Data.Set as Set
+import qualified Data.Map as Map
+import           Data.Map (Map)
+
+import           Control.Monad (when, unless)
+import           Control.Monad.Class.MonadSTM
+import           Control.Monad.Class.MonadThrow
+import           Control.Monad.Class.MonadTime
+import           Control.Exception (assert)
+
+import           Ouroboros.Network.Block
+
+import           Ouroboros.Network.Protocol.BlockFetch.Type
+import           Network.TypedProtocol.Core
+import           Network.TypedProtocol.Pipelined
+
+import           Ouroboros.Network.BlockFetch.Types
+                   ( FetchClientStateVars(..), PeerFetchStatus(..)
+                   , PeerFetchInFlight(..), initialPeerFetchInFlight
+                   , SizeInBytes
+                   , FetchRequest(..)
+                   , newTFetchRequestVar, takeTFetchRequestVar )
+import           Ouroboros.Network.BlockFetch.DeltaQ
+                   ( PeerGSV(..), PeerFetchInFlightLimits(..)
+                   , calculatePeerFetchInFlightLimits )
+
+
+data FetchClientPolicy header block m =
```

Comment:
It would be nice to document the `FetchClientPolicy`, at least the `addFetchedBlock` field.  Can be in a future PR.

Reply:
It's the same as the `addFetchedBlock` in the top level `BlockFetch` API. But yes, there's a lot of things that need to be better documented. I'll do a pass and put that in another PR.